retrieval/scrape.go

```diff
@@ -807,8 +879,11 @@ func (sl *scrapeLoop) reportStale(start time.Time) error {
 }
 
 func (sl *scrapeLoop) addReportSample(app storage.Appender, s string, t int64, v float64) error {
-	ref, ok := sl.refCache[s]
+	// Suffix s with the invalid \xff unicode rune to avoid collisions
+	// with scraped metrics.
+	s2 := s + "\xff"
```

Comment:
Not sure if I understand all the subtleties here, but isn't it supposed to collide if a user chooses to name a metric e.g. `up`?

At the moment that just leads to the synthetic metrics to not be recorded, due to `sample with repeated timestamp but different value`.

Do I understand correctly that that would still be the case with this change, but the only change would be that the user-supplied `up` series would get a stale marker written when it disappears (instead of the series staying non-stale and a synthetic `up` being written instead)?

Apologies if that's all wrong, I'm starting to have difficulty fitting all the relationships between the different caches and refs etc. into my head...